AWSIoT/Internal/AWSIoTMQTTClient.m

```diff
@@ -524,9 +531,16 @@ - (void)initWebSocketConnectionForURL:(NSString *)urlString {
     }
     
     //Setup userName if metrics are enabled
-    NSString *username;
+    NSMutableString *username;
     if (self.isMetricsEnabled) {
-        username = [NSString stringWithFormat:@"%@%@", @"?SDK=iOS&Version=", SDK_VERSION];
+        username = [NSMutableString stringWithFormat:@"%@%@", @"?SDK=iOS&Version=", SDK_VERSION];
```

Comment:
Code duplication - refactor this into an internal function

Reply:
Done.